lib/minitest/assertions.rb

```diff
@@ -170,6 +170,7 @@ def assert_empty obj, msg = nil
     # See also: Minitest::Assertions.diff
 
     def assert_equal exp, act, msg = nil
+      raise ArgumentError, "use assert_nil if expecting nil" if exp.nil?
```

Comment:
I tried that, but it messed up all the tests that count "number of assertions".  Also, I'm not sure if this is an `ArgumentError` or an actual test failure.